tests/test_format_message.py

```diff
@@ -28,4 +28,26 @@ def test_format_triggers(self):
 
         """)
         self.reply("hi there", "hi there")
-        self.reply("hi  here", "hi here")
\ No newline at end of file
+        self.reply("hi  here", "hi here")
+
+    def test_invalid_character_raise_exception(self):
+        self.assertRaises(Exception, self.new, """
+                                + $hello
+                                - hi
+                            """)  # This test passes with `match`, which only check at the beginning
+        self.assertRaises(Exception, self.new, """
+                        + hello$
+                        - hi
+                    """)   # This test does not pass because the beginning is good, no $
+        self.assertRaises(Exception, self.new, """
+                            > topic Greetings
+                                + hello
+                                - hi
+                            <topics
+                            """)
+        self.assertRaises(Exception, self.new, """
+                                    > object hash %perl
+                                      my ($rs, $args) = @_;
+                                      my $method = shift @{$args};
+                                    <object
+                                    """)  # Test for character violation in object, no %
```

Comment:
I don't like how all these blocks of code are formatted. The indentations are all over the place.

Maybe something like this instead:

```python
    def test_invalid_character_raise_exception(self):
        # This test passes with `match`, which only check at the beginning
        self.assertRaises(Exception, self.new, """
            + $hello
            - hi
        """)

        # etc for all the other tests.
```